plugins/by-name/alpha/default.nix

```diff
@@ -137,75 +123,113 @@ in
+          "██╔██╗ ██║██║ ╚███╔╝ ██║   ██║██║██╔████╔██║"
+          "██║╚██╗██║██║ ██╔██╗ ╚██╗ ██╔╝██║██║╚██╔╝██║"
+          "██║ ╚████║██║██╔╝ ██╗ ╚████╔╝ ██║██║ ╚═╝ ██║"
+          "╚═╝  ╚═══╝╚═╝╚═╝  ╚═╝  ╚═══╝  ╚═╝╚═╝     ╚═╝"
+        ];
+        opts = {
+          position = "center";
+          hl = "Type";
+        };
+      }
+      {
+        type = "padding";
+        val = 2;
+      }
+      {
+        type = "group";
+        val = [
           {
-            enable = lib.mkOverride 1490 true;
-          };
+            type = "button";
+            val = "  New file";
+            on_press = nestedLiteralLua "function() vim.cmd[[ene]] end";
+            opts.shortcut = "n";
+          }
+          {
+            type = "button";
+            val = " Quit Neovim";
+            on_press = nestedLiteralLua "function() vim.cmd[[qa]] end";
+            opts.shortcut = "q";
+          }
+        ];
+      }
+      {
+        type = "padding";
+        val = 2;
+      }
+      {
+        type = "text";
+        val = "Inspiring quote here.";
+        opts = {
+          position = "center";
+          hl = "Keyword";
+        };
+      }
+    ];
+  };
+
+  extraOptions = {
+    theme = defaultNullOpts.mkStr' {
+      description = ''
+        The theme to use for the dashboard.
+      '';
+      example = "dashboard";
+      apply =
+        value: if builtins.isString value then mkRaw ''require("alpha.themes.${value}").config'' else value;
+    };
+  };
 
-      extraPlugins = [ cfg.package ];
+  optionsRenamedToSettings = [
+    "opts"
+    "layout"
+  ];
 
-      assertions = lib.nixvim.mkAssertions "plugins.alpha" [
+  extraConfig =
+    cfg:
+    let
+      layoutDefined = cfg.settings.layout != null;
+      themeDefined = cfg.theme != null;
+    in
+    {
+      assertions = mkAssertions "plugins.alpha" [
         {
-          assertion = themeDefined || layoutDefined;
+          assertion = layoutDefined || themeDefined;
           message = ''
             You have to either set a `theme` or define some sections in `layout`.
           '';
         }
         {
-          assertion = !(themeDefined && layoutDefined);
+          assertion = !(themeDefined && cfg.settings != { });
           message = ''
             You can't define both a `theme` and custom options.
             Set `plugins.alpha.theme = null` if you want to configure alpha manually using the `layout` option.
           '';
         }
       ];
 
-      extraConfigLua =
-        let
-          setupOptions =
-            if themeDefined then
-              cfg.theme
-            else
-              (with cfg; {
-                inherit layout opts;
-              });
-        in
-        ''
-          require('alpha').setup(${lib.nixvim.toLuaObject setupOptions})
-          require('alpha.term')
-        '';
+      plugins.alpha.luaConfig.content = ''
+        require('alpha').setup(${toLuaObject (if themeDefined then cfg.theme else cfg.settings)});
```

Comment:
If we're calling `setup` manually, we need to tell mkNeovimPlugin not to also call `setup`.

I.e. add `callSetup = false` somewhere, perhaps immediately before `extraConfig`?